libs/common/types.hpp

```diff
@@ -112,31 +123,42 @@ namespace iroha {
     return std::string(source.begin(), source.end());
   }
 
-  // Deserialize hex string to array
-  template <size_t size>
-  inline void hexstringToArray(const std::string& string, blob_t<size>& array) {
-    auto bytes = hex2bytes(string);
-    std::copy(bytes.begin(), bytes.end(), array.begin());
-  }
-
   /**
    * Convert string of raw bytes to printable hex string
    * @param str
    * @return
    */
   inline std::string bytestringToHexstring(std::string str) {
-    std::string res(str.size() * 2, 0);
-    uint8_t front, back;
-    auto ptr = str.data();
-    for (uint32_t i = 0, k = 0; i < str.size(); i++) {
-      front = (uint8_t) (ptr[i] & 0xF0) >> 4;
-      back = (uint8_t) (ptr[i] & 0xF);
-      res[k++] = code[front];
-      res[k++] = code[back];
+    std::stringstream ss;
+    ss << std::hex << std::setfill('0');
+    for(const auto &c : str) {
+      ss << std::setw(2) << static_cast<int>(c);
```

Comment:
move `std::setw(2)` out of loop:
```
inline std::string bytestringToHexstring(std::string str) {
    std::stringstream ss;
    ss << std::hex << std::setfill('0') << std::setw(2);
    for(const auto &c : str) {
      ss << static_cast<int>(c);
    }
    return ss.str();
  }
```

Reply:
http://en.cppreference.com/w/cpp/io/manip/setw
> The width property of the stream will be reset to zero (meaning "unspecified") if any of the following functions are called:
>      Output 
>      Overloads 1-7 of basic_ostream::operator<<() (at Stage 3 of num_put::put())
>        operator<<(basic_ostream&, char) and operator<<(basic_ostream&, char*)
 >       operator<<(basic_ostream&, basic_string&) 

So it's not possible to move it out of loop.